eng/versioning/verify_release_set.ps1

```diff
@@ -0,0 +1,97 @@
+$missingLibraries = @()
+function Write-Error-With-Color([string]$msg)
+{
+    Write-Host "$($msg)" -ForegroundColor Red
+}
+
+Write-Host "ServiceDirectory=$($ServiceDirectory)"
+Write-Host "ArtifactsList:"
+$ArtifactsList | Format-Table -Property Name, GroupId | Out-String | Write-Host
+foreach($artifact in $ArtifactsList) {
+    $libraryReleaseList += ,($artifact.groupId + ":" + $artifact.name)
+}
+
+foreach($artifact in $ArtifactsList) {
+    $script:FoundPomFile = $false
+    $inputGroupId = $artifact.groupId
+    $inputArtifactId = $artifact.name
+
+    # It's unfortunate that we have to find the POM file this way for a given group/artifact.
+    # The reason is because an sdk/<area> should have subdirectories named for each artifact but
+    # that's not always the case so we need to discover.
+    Get-ChildItem -Path $ServiceDirectory -Filter pom*.xml -Recurse -File | ForEach-Object {
+        $pomFile = $_.FullName
+        $xmlPomFile = New-Object xml
+        $xmlPomFile.Load($pomFile)
+        if (($xmlPomFile.project.groupId -eq $inputGroupId) -and ($xmlPomFile.project.artifactId -eq $inputArtifactId)) {
+            $script:FoundPomFile = $true
+            # Verify that each current dependency is contained within the list of libraries to be released
+            foreach($dependencyNode in $xmlPomFile.GetElementsByTagName("dependency"))
+            {
+                $artifactId = $dependencyNode.artifactId
+                $groupId = $dependencyNode.groupId
+                $versionNode = $dependencyNode.GetElementsByTagName("version")[0]
+
+                $scopeNode = $dependencyNode.GetElementsByTagName("scope")[0]
+                if ($scopeNode -and $scopeNode.InnerText.Trim() -eq "test")
+                {
+                    continue
+                }
+                # if there is no version update tag for the dependency then fail
+                if ($versionNode.NextSibling -and $versionNode.NextSibling.NodeType -eq "Comment")
+                {
+                    $versionUpdateTag = $versionNode.NextSibling.Value.Trim()
+                    if ($versionUpdateTag -match ";current}")
+                    {
+                        $librayToVerify = $groupId + ":" + $artifactId
+                        if ($libraryReleaseList -notcontains $librayToVerify) {
+                            if ($missingLibraries -notcontains $librayToVerify) {
+                                $missingLibraries += ,$librayToVerify
+                            }
+                            $script:FoundError = $true
+                            Write-Error-With-Color "Error: $($pomFile) contains a current dependency, groupId=$($groupId), artifactId=$($artifactId), which is not the list of libraries to be released."
+                        }
+                    }
+                    continue
+                }
+            }
+        } else {
+            return
+        }
+    }
+
+    if (-Not $script:FoundPomFile) {
```

Comment:
nit: I just like `!` over `-not`
```suggestion
    if (!$script:FoundPomFile) {
```

Reply:
Okay. This will be updated with the next commit.